src/ripple/nodestore/Backend.h

```diff
@@ -56,6 +56,13 @@ class Backend
     */
     virtual void open(bool createIfMissing = true) = 0;
 
+    /** Open the backend.
+        @param createIfMissing Create the database files if necessary.
+        @param salt Deterministic salt used to create a backend.
+        This allows the caller to catch exceptions.
+    */
+    virtual void open(bool createIfMissing, boost::optional<uint64_t> salt) {}
```

Comment:
This isn't ideal: it allows an implementation detail to "bleed" into the API, but I don't quite have a better idea on how to handle it.

At a _minimum_ the default implementation of this function, here in `Backend` should throw an `std::runtime_error` exception, with an appropriate error message that explains that the specific backend does not support deterministic salts.

Reply:
Done